x-pack/solutions/security/plugins/security_solution/public/entity_analytics/pages/entity_analytics_privileged_user_monitoring_page.tsx

```diff
@@ -100,11 +102,26 @@ export const EntityAnalyticsPrivilegedUserMonitoringPage = () => {
   const { initPrivilegedMonitoringEngine } = useEntityAnalyticsRoutes();
   const [state, dispatch] = useReducer(reducer, initialState);
 
-  const { sourcererDataView: oldSourcererDataView } = useSourcererDataView();
+  const {
+    indicesExist: oldIndicesExist,
+    loading: oldIsSourcererLoading,
+    sourcererDataView: oldSourcererDataViewSpec,
+  } = useSourcererDataView();
   const newDataViewPickerEnabled = useIsExperimentalFeatureEnabled('newDataViewPickerEnabled');
+  const { dataView, status } = useDataView();
   const { dataViewSpec } = useDataViewSpec();
 
-  const sourcererDataView = newDataViewPickerEnabled ? dataViewSpec : oldSourcererDataView;
+  const isSourcererLoading = useMemo(
+    () => (newDataViewPickerEnabled ? status !== 'ready' : oldIsSourcererLoading),
```

Comment:
what if the status is `error`?

Reply:
The page will load normally, but many functionalities won't work correctly. 

I copied and pasted this from other pages. Most security solution pages have the same issue. Most of them have the some form of `newDataViewPickerEnabled` logic duplicated.